packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { useCallback, SetStateAction, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { transparentize } from 'polished';
+import facepaint from 'facepaint';
+import Portal from '@leafygreen-ui/portal';
+import Icon, { Size } from '@leafygreen-ui/icon';
+import { useEventListener } from '@leafygreen-ui/hooks';
+import { uiColors } from '@leafygreen-ui/palette';
+import { css, cx, keyframes } from '@leafygreen-ui/emotion';
+
+export const ModalSize = {
+  Small: 'small',
+  Default: 'default',
+  Large: 'large',
+} as const;
+
+export type ModalSize = typeof ModalSize[keyof typeof ModalSize];
+
+// breakpoints for different screen sizes
+const small = '767px'; // mobile screens, from 0px - 767px
+const medium = '768px'; // tablet screens, from 768px - 1024px
+const large = '1025px'; // laptops/desktop screens, from 1025px and above
+
+export const mq = facepaint([
+  `@media only screen and (max-width: ${small})`,
+  `@media only screen and (min-width: ${medium})`,
+  `@media only screen and (min-width: ${large})`,
+]);
+
+const defaultSpacing = 18;
+
+const fadein = keyframes`
+  from {
+    margin-top: 50px;
+    opacity: 0;
+  }
+  to {
+    margin-top: 0;
+    opacity: 1;
+  }
+`;
+
+const mainContainer = css`
+  animation: fade-in 250ms ease-in-out;
+  background-color: ${transparentize(0.4, uiColors.black)};
+  overflow-y: auto;
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  z-index: 1000;
+  animation: ${fadein} 300ms ease-in-out;
+}`;
+
+const scrollContainer = css`
+  position: absolute;
+  top: 0;
+  right: 0;
+  left: 0;
+  overflow-y: auto;
+  display: flex;
+  justify-content: center;
+  z-index: 1000;
+`;
+
+const modalContentStyle = css`
+  margin: ${defaultSpacing}px auto;
+  padding: 36px;
+  color: ${uiColors.gray.dark3};
+  background-color: ${uiColors.white};
+  border: 1px solid ${uiColors.gray.light3};
+  border-radius: 3px;
+  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
+  min-width: inherit;
+  max-width: inherit;
+  position: relative;
+  pointer-events: all;
+  z-index: 1000;
```

Comment:
I really don't like arbitrary z-indexes. Can you explain why `1000` is needed, or why we need this at all?

Reply:
Additionally, having `z-index: 1000` in 3 places here is a red flag. I'm certain we don't need all instances of this.

Reply:
This is here because when I tested the component in MMS, the side and top navs were not hidden. They have z-indexes of 998 (possibly 997), so I did this to ensure that the backdrop would conceal the side/left nav in MMS. I can add a comment to indicate that

Reply:
Details related to a specific app's implementation should be left to them. if there's not a way currently for them to implement this, we should add a way.

Reply:
so expose a zIndex prop?

Reply:
@DesignerDave ^

Reply:
Only if this needs to be applied to a different element than className.

Reply:
So omit the zIndex 1000 and let MMS handle that with className on their own?